app/presenters/linked_item_presenter.rb

```diff
@@ -44,4 +49,17 @@ def web_url(item)
 
     Plek.current.website_root + item.base_path
   end
+
+  def add_parents_recursively(parent)
+    json = {
+      content_id: parent.content_id,
+      title: parent.title,
+      base_path: parent.base_path
```

Comment:
It'd be better to create a new instance of this class recursively for each ancestor.